### PR TITLE
Creating interface for commands and adding is-yarn-global dependency

### PR DESCRIPTION
**Pull request by ftonato**
Screen-Recording-2020-08-19-at-23 17 29

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -71,5 +70,7 @@ import { DEV_DEPENDENCIES } from './lib/dependencies'
     `${packageManager} ${devInstall} ${DEV_DEPENDENCIES.join(' ')}`,
     projectDir
   )
+
   spinner.stop()
+  console.log(`👟 ${functionName}'s builded :)`)
```
Comment (atilafassina) [inline src/index.ts]:
```suggestion
   console.log(`👟  ${functionName}’s built`)
```

Comment (atilafassina) [reply src/index.ts]: 2 spaces between emoji and characters help for readability in some terminals

File: src/lib/questions.ts [hunk 2]
```diff
@@ -37,14 +38,15 @@ export default ({ cwd }: { cwd: string }): PromptObject[] => [
...
+        disabled: !isYarnGlobal()
       },
       {
         title: 'NPM',
         description: 'Use NPM as the package manager',
         value: 'npm',
       },
     ],
-    initial: 0,
+    initial: isYarnGlobal() ? 0 : 1,
```
Comment (atilafassina) [inline src/lib/questions.ts:49]: I checked the code on this package, are you sure this does not give false negatives on WSL ?

File: src/lib/utils/command.ts [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
-    default:
-      return '--help'
+interface ICommandOptions {
+  [key: string]: any;
+}
+
+export function command(pkgManager: string, cmd: string): string {
+  let _cmd = '--help'
+  
+  const options: ICommandOptions = {
```
Comment (atilafassina) [inline src/lib/utils/command.ts:8]: all commands are for global installation here 😬

**Review by atilafassina — CHANGES_REQUESTED** [review]
Absolutely no need to be sorry for.
Thank you very much for this PR!!!! 🏆

I forced pushed a commit on top of my last one before this PR.
I'm afraid you will need some rebasing to dodge the conflicts. Sorry.

File: src/lib/questions.ts (hunk 2 above)
Comment (ftonato) [reply src/lib/questions.ts:49]: unfortunately, I'm not sure = (

File: src/lib/utils/command.ts [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+    'GLOBAL_INSTALL': pkgManager === 'yarn' ? 'global add --silent' : 'i -g',
+    'LOCAL_INSTALL': pkgManager === 'yarn' ? 'global add --silent' : 'i -g',
```
Comment (atilafassina) [inline src/lib/utils/command.ts]:
```suggestion
    'LOCAL_INSTALL': pkgManager === 'yarn' ? 'add --silent' : 'i',
```

File: src/lib/utils/command.ts [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
+    'DEV_INSTALL': pkgManager === 'yarn' ? 'global add --silent' : 'i -g',
```
Comment (atilafassina) [inline src/lib/utils/command.ts]:
```suggestion
    'DEV_INSTALL': pkgManager === 'yarn' ? 'add -D --silent' : 'i -D',
```

**Review by atilafassina — CHANGES_REQUESTED** [review]

**Review by atilafassina — APPROVED** [review]
Awesome!!

Thanks a lot 💚

**Pull request merged**